lib/queue.js

```diff
@@ -505,7 +505,14 @@ Queue.prototype.processStalledJob = function(job){
           if(!isMember){
             _this.emit('stalled', job);
             return _this.processJob(job, true);
+          } else {
+            return job.releaseLock(_this.token);
           }
+        }).catch(function(err) {
+          // Any uncaught error will come here. We'll ensure that the job lock is released and rethrow
+          // the error so it bubbles normally
+          job.releaseLock(_this.token);
+          throw err;
         });
```

Comment:
Hey @manast, sorry for the delay.

We will be in a similar situation, because this `finally` will remove attempt to remove a lock in the event that `processJob` runs, since `processJob` already releases the lock, we'll be releasing twice.

I think we should still do it in `catch` because `catch` will run on a codepath where `processJob` does not release the lock, thus only releasing the lock once.